solana-ibc-story.md

```diff
@@ -1,65 +1,66 @@
-talks about sub-second finality times in its abstract and [Solana
+talks about sub-second finality times in its abstract and the [Solana
 website](https://solana.com/) advertises a 400-millisecond block time.
 
 To achieve such impressive speeds, the blockchain had to introduce
-various limits ranging from ~1 KB transaction size through maximum
-account size to relatively small heap during smart contract execution.
-For majority of programs on the blockchains those limits aren’t
-a problem.  An NFT contract needs only a couple hundreds bytes of
+various limits ranging from a ~1 KB transaction size through maximum
+account size to a relatively small heap during smart contract execution.
+For a majority of programs on the blockchain, those limits aren’t
+a problem.  For example, an NFT contract needs only a couple hundred bytes of
 transaciton data and fits easily within the ‘relatively small’ heap.
 
-Implementing [the Inter-Blockchain Communication protocol
-(IBC)](https://www.ibcprotocol.dev/) is however not a trivial task.
+Implementing [the Inter-Blockchain Communication (IBC) protocol](https://www.ibcprotocol.dev/) is however not a trivial task.
 With all the encapsulation, block header signatures, state proofs
-etc. which IBC core needs to handle, the contract can easily go past
-those limits.  In this post I go through some of the things we had to
-do to bring IBC onto Solana.
+etc. that the IBC core needs to handle, the contract can easily surpass
+those limits.  In this post, I go through some of the work we had to
+do to overcome various challenges and successfully bring IBC onto Solana.
 
 
-## The Guest blockchain
+## The Guest blockchain and scope of the present article
 
-Fist of, Solana doesn’t meet IBC’s host requirements listed in [ICS 24
+Fist of all, Solana doesn’t meet each of IBC’s host requirements listed in the [ICS 24
 document](https://github.com/cosmos/ibc/blob/main/spec/core/ics-024-host-requirements/README.md).
-It offers neither state proofs nor introspection.  The former is
-required to create membership and non-membership proofs.  The latter
-to preform IBC handshake when establishing an IBC connection.  This is
-why we’ve ended up creating [the Guest
+Specifically, it offers neither state proofs nor introspection.  The former is
+required to create membership and non-membership proofs.  The latter is needed
+to perform an IBC handshake when establishing an IBC connection.  This is
+why we’ve ended up creating [the guest
 blockchain](https://docs.picasso.network/technology/ibc/solana/#ibc-requirements--the-need-for-a-guest-blockchain)
-which runs atop Solana.  While interesting topic by itself, this post
-isn’t be about it.
+which runs atop Solana.  While an interesting topic by itself, this post
+is not about the guest blockchain, as the guest blockchain has been detailed [here](https://research.composable.finance/t/crossing-the-cross-blockchain-interoperability-chasm/33).
```

Comment:
Would this be better as 

```suggestion
is not about the guest blockchain, as it has been detailed [here](https://research.composable.finance/t/crossing-the-cross-blockchain-interoperability-chasm/33).
```

Otherwise we repeat ‘the guest blockchain’ three times in close succession.